parsers/cxx/cxx_parser.c

```diff
@@ -463,8 +464,24 @@ static bool cxxParserParseEnumStructClassOrUnionFullDeclarationTrailer(
 	if(uKeywordState & CXXParserKeywordStateSeenTypedef)
 		cxxParserExtractTypedef(g_cxx.pTokenChain,true);
 	else
+	{
 		cxxParserExtractVariableDeclarations(g_cxx.pTokenChain,0);
 
+		/*
+		   Skip initializer in
+
+		   struct foo { ... } x = { ... };
+
+		   if we are at --------^.
+		*/
+		if(cxxTokenTypeIs(g_cxx.pToken,CXXTokenTypeAssignment) &&
+		   (!cxxParserParseUpToOneOf(CXXTokenTypeEOF | CXXTokenTypeSemicolon)))
+		{
+			CXX_DEBUG_LEAVE_TEXT("Failed to parse up to EOF/semicolon");
+			return false;
+		}
+	}
+
```

Comment:
CXXTokenTypeAssignment should be probably handled at all exit points.
If Typedef has been seen, its an error (unless I'm missing something), but we might just be tolerant and handle it anyway.